plugin/core/sessions.py

```diff
@@ -1401,14 +1401,18 @@ def m_workspace_applyEdit(self, params: Any, request_id: Any) -> None:
     def m_workspace_semanticTokens_refresh(self, params: Any, request_id: Any) -> None:
         """handles the workspace/semanticTokens/refresh request"""
         self.send_response(Response(request_id, None))
-        # TODO should we send a request only for the active view, or also for all other SessionBuffers of the Session?
-        active_view = sublime.active_window().active_view()
-        if not active_view:
-            return
-        session_view = self.session_view_for_view_async(active_view)
-        if not session_view:
-            return
-        session_view.session_buffer.do_semantic_tokens_async(active_view)
+        selected_sheets = self.window.selected_sheets()
+        for sheet in self.window.sheets():
+            view = sheet.view()
+            if not view:
+                continue
+            sv = self.session_view_for_view_async(view)
+            if not sv:
+                continue
+            if sheet in selected_sheets:
+                sv.session_buffer.do_semantic_tokens_async(view)
```

Comment:
Just a note since that's not an issue with your changes but the type of `sv.session_buffer` is `Any` so we don't have any type checking here.

Reply:
So that means I cannot call `do_semantic_tokens_async` on `sv.session_buffer`? I guess I will wait until your other PR is merged and see what I have to do then.
This line needs a little tweak anyway, because ideally we don't want to request tokens multiple times here for a certain buffer, which would happen now if a view is split and more than one of the split views corresponding to the single buffer are selected at the same time.

Reply:
You would then need to declare that method on `SessionBufferProtocol`.